pkg/sanity/volume.go

```diff
@@ -329,43 +510,104 @@ var _ = Describe("Volume [Volume Tests]", func() {
+
+			By("Attaching a volume before mounting")
+
+			str, err := volumedriver.Attach(volumeID, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(str).NotTo(BeEmpty())
+
+			By("Attaching mounting the volume to mount Path")
+
+			err = volumedriver.Mount(volumeID, mountPath, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Unmounting the volume successfully")
+			err = volumedriver.Unmount(volumeID, mountPath, nil)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Describe("Volume Update", func() {
+
+		var (
+			numVolumesBefore int
+			numVolumesAfter  int
+			volumeID         string
+		)
+
+		BeforeEach(func() {
+			var err error
+			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, make(map[string]string))
+			numVolumesBefore = len(volumes)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			var err error
+			err = volumedriver.Delete(volumeID)
+			Expect(err).ToNot(HaveOccurred())
+
 			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, nil)
+			Expect(err).ToNot(HaveOccurred())
 			numVolumesAfter = len(volumes)
+		})
 
-			Expect(numVolumesAfter).To(Equal(numVolumesBefore + 1))
+		It("Should update successfully", func() {
 
-			By("Doing a Volume Set to mount the volume")
+			By("Creating the volume")
 
-			// req := &api.VolumeSetRequest{
-			// 	Options: map[string]string{},
-			// 	Action: &api.VolumeStateAction{
-			// 		Attach:    api.VolumeActionParam_VOLUME_ACTION_PARAM_ON,
-			// 		Mount:     api.VolumeActionParam_VOLUME_ACTION_PARAM_ON,
-			// 		MountPath: "/mnt/testmountvolume",
-			// 	},
-			// 	Locator: &api.VolumeLocator{Name: vr.GetLocator().GetName()},
-			// 	Spec:    &api.VolumeSpec{Size: vr.GetSpec().GetSize()},
-			// }
+			var err error
 
-			_, err = volumedriver.Attach(volumeID, nil)
-			Expect(err).NotTo(HaveOccurred())
-			err = volumedriver.Mount(volumeID, "/mnt", nil)
+			var size uint64 = 5
+			vr := &api.VolumeCreateRequest{
+				Locator: &api.VolumeLocator{
+					Name: "create-to-attach",
+					VolumeLabels: map[string]string{
+						"class": "attach-class",
+					},
+				},
+				Source: &api.Source{},
+				Spec: &api.VolumeSpec{
+					Size:    size,
+					HaLevel: 1,
+					Format:  api.FSType_FS_TYPE_EXT4,
+				},
+			}
 
-			//err = volumedriver.Set(volumeID, req.GetLocator(), req.GetSpec())
+			volumeID, err = volumedriver.Create(vr.GetLocator(), vr.GetSource(), vr.GetSpec())
 
 			Expect(err).NotTo(HaveOccurred())
+			Expect(volumeID).ToNot(BeNil())
 
-			By("Inspecting the volume and checking attached_on field ")
+			By("Checking if volume created successfully with the provided params")
+			testIfVolumeCreatedSuccessfully(volumedriver, volumeID, numVolumesBefore, vr)
 
-			volumes, err = volumedriver.Inspect([]string{volumeID})
+			By("Updating the volume spec. ")
+			newSize := 5
 
+			set := &api.VolumeSetRequest{
+				Locator: vr.GetLocator(),
+				Spec: &api.VolumeSpec{
+					Size:   uint64(newSize),
```

Comment:
not all drivers are going to support setting the size field. We might need to check for capabilities first and then do it. May be start with just setting the labels. We can have a separate test that does a resize.